sys/shell/commands/sc_vfs.c

```diff
@@ -227,6 +228,121 @@ static int _read_handler(int argc, char **argv)
     return 0;
 }
 
+static int _write_handler(int argc, char **argv)
+{
+    char *w_buf;
+    char buf[16];
+    size_t nbytes = 0;
+    size_t nb_str = 0;
+    char *path = argv[1];
+    int ascii = 0;
+    int flag = O_CREAT;
+    if (argc < 2) {
+        puts("vfs write: missing file name");
+        return 1;
+    }
+
+    if (argc < 3) {
+        puts("vfs write: missing format");
+        return 1;
+    }
+    if (strcmp(argv[2], "ascii") == 0) {
+        ascii = 1;
+    }
+    else if (strcmp(argv[2], "hex") == 0) {
+        ascii = 0;
+    }
+    else {
+        printf("vfs write: unknown format: %s\n", argv[2]);
+        return 1;
+    }
+
+    if (argc < 4) {
+        puts("vfs write: missing <a|o> flag");
+        return 1;
+    }
+    if (strcmp(argv[3], "a") == 0) {
+        flag |= O_WRONLY | O_APPEND;
+    }
+    else if (strcmp(argv[3], "o") == 0) {
+        flag |= O_WRONLY;
+    }
+    else {
+        printf("vfs write: invalid flag %s\n", argv[3]);
+        return 1;
+    }
+
+    if (argc < 5) {
+        puts("vfs write: missing data");
+        return 1;
+    }
+    w_buf = argv[4];
+    nbytes = strlen(w_buf);
+    if (!ascii) {
+        nb_str = argc - 4;
+    }
+
+    int res;
+    res = vfs_normalize_path(path, path, strlen(path) + 1);
+    if (res < 0) {
+        _errno_string(res, (char *)buf, sizeof(buf));
+        printf("Invalid path \"%s\": %s\n", path, buf);
+        return 5;
+    }
+
+    int fd = vfs_open(path, flag, 0);
+    if (fd < 0) {
+        _errno_string(fd, (char *)buf, sizeof(buf));
+        printf("Error opening file \"%s\": %s\n", path, buf);
+        return 3;
+    }
+
+    if (flag & O_APPEND) {
+        res = vfs_lseek(fd, 0, SEEK_END);
```

Comment:
I just realized that we should make sure that O_APPEND is honored in the FS driver. If O_APPEND is set then all writes should happen as if the seek position was placed at the end of the file beforehand, but reads should still use the normal file position set by lseek. The above line should not be needed if the vfs/fs handles the O_APPEND option properly.